### PR TITLE
fix: False positive on Credo.Check.Refactor.Apply when piped

### PR DESCRIPTION
**Pull request by cheerfulstoic**
An attempt to fix https://github.com/rrrene/credo/issues/993

### CODE REVIEW

File: test/credo/check/refactor/apply_test.exs [hunk 1]
```diff
@@ -107,6 +107,20 @@ defmodule Credo.Check.Refactor.ApplyTest do
...
+        module
+        |> apply(fun, [arg1, arg2])
+      end
+    end
+    """
+    |> to_source_file
+    |> run_check(@described_check)
+    |> refute_issues()
+  end
+
```
Comment (rrrene) [inline test/credo/check/refactor/apply_test.exs:123]: We need to make sure that the cases where we expect issues are still yielding them after this PR is merged:

```suggestion

  test "it should report violation for apply/2 in a pipe" do
    """
    defmodule Test do
      def some_function(fun, arg1, arg2) do
        fun |> apply([arg1, arg2])
      end
    end
    """
    |> to_source_file
    |> run_check(@described_check)
    |> assert_issue()
  end

  test "it should report a violation for apply/3 in a pipe" do
    """
    defmodule Test do
      def some_function(module, arg1, arg2) do
        module |> apply( :fun_name, [arg1, arg2])
      end
    end
    """
    |> to_source_file
    |> run_check(@described_check)
    |> assert_issue()
  end
```

**Review by rrrene — COMMENTED** [review]
Great! Just a small request :+1:

File: test/credo/check/refactor/apply_test.exs (hunk 1 above)
Comment (cheerfulstoic) [reply test/credo/check/refactor/apply_test.exs:123]: Nice catch, thank you!  This led me to a nicer solution where I just called `issue` again with a re-combined ast which looks like an `apply` without a pipe.

**Comment by rrrene** [comment]
This is part of Credo [v1.7.0-rc.1](https://hex.pm/packages/credo/1.7.0-rc.1) :+1:

**Pull request merged**